### PR TITLE
Mc refactors

### PR DESCRIPTION
**Pull request by Mariana-21**
## Spec/Features ##
* Refactor in cart show_spec

## Controllers ## 
* refactor in cart controller created 

## Views ##
* refactor in merchant item new form

### CODE REVIEW

File: spec/features/cart/show_spec.rb [hunk 1]
```diff
@@ -102,21 +102,46 @@
         end
         expect(page).to have_no_content(@tire.name)
       end
+
+      it 'I see a message saying that I cannot go passed the inventory limit' do
```
Comment (perryr16) [inline spec/features/cart/show_spec.rb:106]: nice new test

Comment (foleymichelle9) [reply spec/features/cart/show_spec.rb:106]: looks good!

File: spec/models/cart_spec.rb [hunk 2]
```diff
@@ -63,5 +63,15 @@
 
       expect(@cart.quantity_one?(@giant.id.to_s)).to eq(true)
     end
+
+    it '.inventory_reached?()' do
+      expect(@cart.inventory_reached?(@hippo.id.to_s)).to eq(false)
+
+      3.times do
```
Comment (perryr16) [inline spec/models/cart_spec.rb:70]: this is intelligent!

**Comment by foleymichelle9** [comment]
Approved!

**Pull request merged**